upload_tests/test_s3_uploader.py

```diff
@@ -0,0 +1,105 @@
+#  Drakkar-Software OctoBot-Tentacles-Manager
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import json
+import os
+import shutil
+import zipfile
+
+import aiofiles
+import aiohttp
+import time
+
+import pytest
+
+import octobot_tentacles_manager.api.uploader as uploader_api
+import octobot_tentacles_manager.constants as constants
+
+# All test coroutines will be treated as marked.
+pytestmark = pytest.mark.asyncio
+
+TEST_S3_DIRECTORY = "s3-tests"
+TEST_S3_PREFIX = "/tests/"
+TEST_S3_PATH = "tentacle-manager/"
+TEST_S3_FILE_NAME = "test"
+
+
+@pytest.fixture
+async def s3_tests():
+    if os.path.exists(TEST_S3_DIRECTORY):
+        shutil.rmtree(TEST_S3_DIRECTORY)
+    os.mkdir(TEST_S3_DIRECTORY)
+    yield
+    if os.path.exists(TEST_S3_DIRECTORY):
+        shutil.rmtree(TEST_S3_DIRECTORY)
```

Comment:
should we also remove the uploaded files on s3 ?

Reply:
They are automatically cleaned after 1 day